paddle/fluid/operators/gru_unit_op.cc

```diff
@@ -46,23 +41,45 @@ class GRUUnitOp : public framework::OperatorWithKernel {
     int frame_size = hidden_prev_dims[1];
     int weight_height = weight_dims[0];
     int weight_width = weight_dims[1];
-    PADDLE_ENFORCE_EQ(
-        input_size, frame_size * 3,
-        "The input_size must be 3 times of frame_size in GRUUnitOp.");
+    if (ctx->IsRuntime()) {
+      PADDLE_ENFORCE_EQ(input_size, frame_size * 3,
+                        platform::errors::InvalidArgument(
+                            "The second dimension of Input(Input) must be 3 "
+                            "times of frame_size in GRUUnitOp, but received %d "
+                            "(Input) vs %d (frame_size).",
+                            input_size, frame_size));
+    }
```

Comment:
In compile time, checking `input_size` is  if `input_size` >= 0.

Reply:
Good advice! Done.